pkg/snapstore/s3_snapstore.go

```diff
@@ -38,19 +40,29 @@ const (
 	s3NoOfChunk            int64 = 9999
 	awsCredentialDirectory       = "AWS_APPLICATION_CREDENTIALS"
 	awsCredentialJSONFile        = "AWS_APPLICATION_CREDENTIALS_JSON"
+	// For server-side encryption
+	sseCustomerAlgorithm = "AES256"
 )
 
 type awsCredentials struct {
 	AccessKeyID        string  `json:"accessKeyID"`
 	Region             string  `json:"region"`
 	SecretAccessKey    string  `json:"secretAccessKey"`
+	SSECustomerKey     string  `json:"sseCustomerKey"`
```

Comment:
```suggestion
	SSECustomerKey     string  `json:"sseCustomerKey,omitempty"`
```

Reply:
Made `SSECustomerKey``*string` as well, since all other fields which are optional `*string`. Another reason is that the customer key could be an extremely long string.

Reply:
See [143474b](https://github.com/gardener/etcd-backup-restore/pull/719/commits/143474b29fad72d23aa50acecdbbe53a60529e87)